capa/features/extractors/dnfile/helpers.py

```diff
@@ -196,6 +214,55 @@ def get_dotnet_managed_methods(pe: dnfile.dnPE) -> Iterator[DnMethod]:
             yield DnMethod(token, row.TypeNamespace, row.TypeName, index.row.Name)
 
 
+def get_dotnet_fields(pe: dnfile.dnPE) -> Iterator[DnProperty]:
+    """get fields from TypeDef table"""
+    for row in iter_dotnet_table(pe, "TypeDef"):
+        for index in row.FieldList:
+            token = calculate_dotnet_token_value(index.table.number, index.row_index)
+            yield DnProperty(token, row.TypeNamespace, row.TypeName, index.row.Name)
+
+
+def get_dotnet_property_map(
+    pe: dnfile.dnPE, property_row: dnfile.mdtable.PropertyRow
+) -> Optional[dnfile.mdtable.TypeDefRow]:
+    """get property map from PropertyMap table
+
+    see https://www.ntcore.com/files/dotnetformat.htm
+
+    21 - PropertyMap Table
+        List of Properties owned by a specific class.
+            Parent (index into the TypeDef table)
+            PropertyList (index into Property table). It marks the first of a contiguous run of Properties owned by Parent. The run continues to the smaller of:
+                the last row of the Property table
+                the next run of Properties, found by inspecting the PropertyList of the next row in this PropertyMap table
+    """
+    for row in iter_dotnet_table(pe, "PropertyMap"):
+        for index in row.PropertyList:
+            if index.row.Name == property_row.Name:
+                return row.Parent.row
+    return None
+
+
+def get_dotnet_properties(pe: dnfile.dnPE) -> Iterator[Optional[DnProperty]]:
+    """get property from MethodSemantics table
+
+    see https://www.ntcore.com/files/dotnetformat.htm
+
+    24 - MethodSemantics Table
+        Links Events and Properties to specific methods. For example one Event can be associated to more methods. A property uses this table to associate get/set methods.
+            Semantics (a 2-byte bitmask of type MethodSemanticsAttributes)
+            Method (index into the MethodDef table)
+            Association (index into the Event or Property table; more precisely, a HasSemantics coded index)
+    """
+    for row in iter_dotnet_table(pe, "MethodSemantics"):
+        if row.Method is not None:
+            typedef_row = get_dotnet_property_map(pe, row.Association.row)
+            if typedef_row is not None:
+                token = calculate_dotnet_token_value(row.Method.table.number, row.Method.row_index)
+                yield DnProperty(token, typedef_row.TypeNamespace, typedef_row.TypeName, row.Association.row.Name)
+        yield None
```

Comment:
I just wanted to return `None` in case the `if` checks do not pass and we can't find the property (they don't pass with the current version of dnfile since the `MethodSemantics` table is not completely populated). We could definitely remove them once we do not need the `if` checks.

Reply:
Sure, https://github.com/mandiant/capa/pull/1122/commits/d35cd47b35086336bdf1a94411f370d39f4a6570 changes the return type.